src/sentry/notifications/utils/actions.py

```diff
@@ -5,40 +5,52 @@
 from typing import Any, Literal
 
 
-@dataclass
-class MessageAction:
-    name: str
-
-    # Optional label. This falls back to name.
-    label: str | None = None
+@dataclass(kw_only=True)
+class BaseMessageAction:
+    """
+    Base class used to hold the fields for a notification message action
+    """
 
+    name: str
     type: Literal["button", "select"] = "button"
-
-    # If this is a button type, a url is required.
+    # If the message action is a button type, the url is required
     url: str | None = None
-
-    # If this is a select type, the selected value.
+    # If the message action is a select type, this is the selected value
     value: str | None = None
-
     # Denotes the type of action
     action_id: str | None = None
+    block_id: str | None = None
+    selected_options: Sequence[Mapping[str, Any]] | None = None
 
-    style: Literal["primary", "danger", "default"] | None = None
+    def _get_button_text(self) -> str:
+        return self.name
 
-    # TODO(mgaeta): Refactor this to be provider-agnostic
-    selected_options: Sequence[Mapping[str, Any]] | None = None
-    option_groups: Sequence[Mapping[str, Any]] | None = None
-    block_id: str | None = None
-    elements: Sequence[Mapping[str, Any]] | None = None
+    def get_button(self) -> Mapping[str, Any]:
```

Comment:
this doesn't need to be at the base class, but everything is so hard typed to `MessageAction` that we can't simply move this to the block kit class, because this is really just needed for slack flows